python/ray/data/tests/test_numpy.py

```diff
@@ -130,8 +130,9 @@ def test_numpy_read(ray_start_regular_shared, tmp_path):
     os.mkdir(path)
     np.save(os.path.join(path, "test.npy"), np.expand_dims(np.arange(0, 10), 1))
     ds = ray.data.read_numpy(path, override_num_blocks=1)
-    assert str(ds) == (
-        "Dataset(num_rows=?, schema={data: numpy.ndarray(shape=(1,), dtype=int64)})"
+    assert ds.count() == 10
+    assert ds.schema() == Schema(
+        pa.schema([("data", ArrowTensorType((1,), pa.int64()))])
```

Comment:
Refactor some tests that were missed by https://github.com/ray-project/ray/pull/48124